### PR TITLE
Updated (ulb, udb, translationWords) to latest resources

### PR DESCRIPTION
**Pull request by mannycolon**
#### This pull request addresses:
- Updated (ulb, udb, translationWords) to latest resources.
- Added a migration action that will ensure we delete the user's resources folder every time a new tC version is released.

#### How to test this pull request:
- Temporary change the app version in the package.json (This will trigger the resources migration code that deletes your local resources folder and regenerates it again for you)
- Verify you see version 10 for ulb and udb and version 6 for translationWords.
- basically, test both tools and multiple projects and make sure it renders all resources.
- try loading all bibles available in the scripture pane.

---
This change is [Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/2387)

### CODE REVIEW

File: scripts/resources/helpers/resourcesHelpers.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import yaml from 'yamljs';
+
+/**
+ * 
+ * @param {array} bibles 
+ * @param {String} extractedFilePath 
+ * @param {String} RESOURCE_OUTPUT_PATH 
+ */
+export function generateBibles(bibles, extractedFilePath, RESOURCE_OUTPUT_PATH) {
+  console.log('\x1b[33m%s\x1b[0m', "Generating tC compatible resource data structure ...")
```
Comment (RoyalSix) [inline scripts/resources/helpers/resourcesHelpers.js]: should this be here? If its necessary for tracking progress thats fine.

File: scripts/resources/helpers/resourcesHelpers.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    // get parsed book
+    const chapters = Object.keys(jsonBibleBook);
+    chapters.forEach(chapterNumber => {
+      let arrayValue = parseInt(chapterNumber, 10);
+      // only allow chapter numbers to generate bible (no strings)
+      if (typeof arrayValue === "number" && !isNaN(arrayValue)) {
+        let fileName = chapterNumber + ".json";
+        let bibleId = jsonBibleBook.headers["toc3"].toLowerCase().replace(" ", "");
+        console.log(bibleId)
```
Comment (RoyalSix) [inline scripts/resources/helpers/resourcesHelpers.js]: console.log

File: src/js/actions/ResourcesActions.js [hunk 3]
```diff
@@ -1,15 +1,13 @@
+/* eslint-disable no-console */
```
Comment (RoyalSix) [inline src/js/actions/ResourcesActions.js:1]: What does this do?

File: package.json [hunk 4]
```diff
@@ -53,10 +58,7 @@
   "dependencies": {
     "adm-zip": "^0.4.7",
     "axios": "^0.16.2",
-    "babel-cli": "^6.24.1",
     "babel-plugin-transform-async-to-generator": "^6.24.1",
-    "babel-plugin-transform-object-rest-spread": "^6.23.0",
-    "babel-preset-env": "^1.6.0",
```
Comment (klappy) [inline package.json:59]: Try running `npm run build` and test the build after making this change. I can't remember if the babel plugins/presets are required in Production with Electron-Forge.

File: src/js/helpers/LoadHelpers.js [hunk 5]
```diff
@@ -239,7 +238,7 @@ export function getParams(path, manifest) {
     }
 
     let params = {
-        'originalLanguagePath': ORIGINAL_LANGUAGE_PATH
+        'originalLanguagePath': ''
```
Comment (klappy) [inline src/js/helpers/LoadHelpers.js:241]: I think I'm missing something but not sure how the path can be ''

File: src/js/helpers/LoadHelpers.js [hunk 6]
```diff
@@ -438,7 +438,7 @@ export function getUSFMParams(projectPath, manifest) {
     else if (manifest.ts_project) bookAbbr = manifest.ts_project.id;
     let direction = manifest.target_language.direction;
     let params = {
-        originalLanguagePath: ORIGINAL_LANGUAGE_PATH,
+        originalLanguagePath: '',
```
Comment (klappy) [inline src/js/helpers/LoadHelpers.js:441]: same question here about how path can be ''

File: src/js/actions/ResourcesActions.js (hunk 3 above)
Comment (mannycolon) [reply src/js/actions/ResourcesActions.js:1]: it disables the Eslint messages for console logs

File: scripts/resources/helpers/stringsHelpers.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
+export function getResourceType(resourceId) {
```
Comment (klappy) [inline scripts/resources/helpers/stringsHelpers.js:1]: Can we rename this file to something more appropriate? Like `resourceHelpers` or `resourceTypeHelpers`?

File: src/js/reducers/settingsReducer.js [hunk 8]
```diff
@@ -7,9 +7,9 @@ const initialState = {
     developerMode: false,
     csvSaveLocation: null,
     online: true,
-    toolsSettings: {},
     onlineMode: false
-  }
+  },
+  toolsSettings: {}
```
Comment (klappy) [inline src/js/reducers/settingsReducer.js:12]: Didn't think of moving this up but that's a really good move.

**Comment by klappy** [comment]
Nice Job! I wasn't expecting testing to go so well!!!

**Comment by klappy** [comment]
The only issue I see shouldn't slow down our merging of this, but needs to be discussed before it gets to QA.

When loading in tW for Titus, the Scripture Pane doesn't highlight the quote. 
The reason I say it shouldn't be addressed now, is that all of that changes once we use Greek as the source for tW, we highlight based on that and we shouldn't waste time fixing English. It would be doing the work twice.

**Review by klappy — COMMENTED** [review]
Looks good! Just a few questions.

File: src/js/helpers/LoadHelpers.js (hunk 6 above)
Comment (mannycolon) [reply src/js/helpers/LoadHelpers.js:441]: The params variable is being removed and ORIGINAL_LANGUAGE_PATH was undefined anyway.

File: src/js/helpers/LoadHelpers.js (hunk 5 above)
Comment (mannycolon) [reply src/js/helpers/LoadHelpers.js:241]: The params variable is being removed and ORIGINAL_LANGUAGE_PATH was undefined anyway.
